chain/processor.go

```diff
@@ -55,53 +51,14 @@ func (b *StatelessBlock) Execute(
 
 		stateKeys, err := tx.StateKeys(sm)
 		if err != nil {
+			f.Stop()
 			e.Stop()
 			return nil, nil, err
 		}
+		// Fetch keys from disk or check if it's in cache
```

Comment:
I thought about this some more and this will actually be a good bit faster than what we currently do because we'll start the lookup right away rather than waiting for a transaction to be considered executable.

While one could argue this means we are prioritizing disk reads that aren't ready to execute (if a set of keys is blocked serially), it isn't really the case because all but one of those reads will be from a cache.